src/pymodaq/daq_utils/h5exporters.py

```diff
@@ -0,0 +1,227 @@
+        attributes and methods, see definition: h5node_exporter.H5Exporter
+
+        See h5node_exporter.H5txtExporter and h5node_exporter.H5txtExporter for usage examples.
+
+        returns:
+            the exporter class
+        """
+
+        def inner_wrapper(wrapped_class) -> Callable:
+            extension = wrapped_class.FORMAT_EXTENSION
+            #Warn if overriding existing exporter
+            if extension in cls.exporters_registry:
+                logger.warning(f"Exporter for the .{extension} format already exists and will be replaced")
+
+            #Register extension
+            cls.exporters_registry[extension] = wrapped_class
+            cls.file_filters[extension] = wrapped_class.FORMAT_DESCRIPTION
+            #Return wrapped_class
+            return wrapped_class
+
+        #Return decorated function
+        return inner_wrapper
+
+    @classmethod
+    def create_exporter(cls, extension: str):
+        """Factory command to create the exporter object.
+
+            This method gets the appropriate executor class from the registry
+            and instantiates it.
+
+            Args:
+                extension (str): the extension of the file that will be exported
+
+            returns:
+                an instance of the executor created
+        """
+        if extension not in cls.exporters_registry:
+            raise ValueError(f".{extension} is not a supported file format.")
+
+        exporter_class = cls.exporters_registry[extension]
+
+        exporter = exporter_class()
+
+        return exporter
+
+    @classmethod
+    def get_file_filters(cls):
+        """Create the file filters string"""
+        tmplist = [f"{v} (*.{k})" for k,v in cls.file_filters.items()]
+        return ";;".join(tmplist)
+
+class H5Exporter(metaclass=ABCMeta):
+    """Base class for an exporter. """
+
+    # This is to define an abstract class attribute
+    @classmethod
+    @property
+    @abstractmethod
+    def FORMAT_DESCRIPTION(cls):
+        """str: file format description as a short text. eg: text file"""
+        raise NotImplementedError
+
+    @classmethod
+    @property
+    @abstractmethod
+    def FORMAT_EXTENSION(cls):
+        """str: File format extension. eg: txt"""
+        raise NotImplementedError
+
+    def __init__(self):
+        """Abstract Exporter Constructor"""
+        pass
+
+    @abstractmethod
+    def _export_data(self, node, filename) -> None:
+        """Abstract method to save a .h5 node to a file"""
+        pass
+
+@ExporterFactory.register_exporter()
+class H5h5Exporter(H5Exporter):
+    """ Exporter object for saving nodes as single h5 files"""
+
+    FORMAT_DESCRIPTION = "Single node h5 file"
+    FORMAT_EXTENSION = "h5"
+
+    def _export_data(self, node, filename) -> None:
+        """Export an h5 node"""
+        #This should allow to get the base file object
+        basefile = node._v_file
```

Comment:
there is an issue here. For h5 files there are multiple possible backends (hence the h5backend code) and I created a wrapper around those (that I could refactor with a factory by the way...). Because of that the variable *node* here is not a pytables node but a Node object (or child Array classes) from the h5backend module. If you want to access the pytables node you'll have to call node.node (that will return the pytables version of the node if you are using the pytables backend) but would return the h5py version of the node if you use the h5py backend.

using the _v_file property will therefore work only with the pytables backend (not with the h5py or h5pyd). So to make it work quick and dirty and only for pytables, you have to write:
``` 
basefile = node.node._v_file
```
but the best approach would be to add a file method to the Node object (in h5backend) and write a specific call for each backend. thnks ;-)

Reply:
Thanks for the heads up!

Indeed I didn't understand very well this whole back-end deal then. I'll refactor this now

Reply:
I remember also now. This is the only way I found (not too troublesome) to get a node and copy it into a new file. Another solution would be to save an array into a new file and scan and copy all its attributes. That would work for all backend transparently!

Reply:
In the end I managed to deal with it. Not in the most elegant way as it involves direct h5pyd/pytables invocation in the code but that will do the trick. In the end I like better a paradigm where backend is entirely abstracted out of the `_export_data` methods even if it involves one or two node tricks.